Invalid MigCluster reference in MigPlan causes SIGSEGV

## Issue Summary
I created a MigPlan resource with out of date MigCluster names and saw the controller crash. We should handle this type of situation gracefully without the controller dying.

**MigPlan with invalid MigCluster refs**
```
apiVersion: migration.openshift.io/v1alpha1
kind: MigPlan
metadata:
  labels:
    controller-tools.k8s.io: "1.0"
  name: migplan-sample
  namespace: mig

  srcMigClusterRef:
    name: myfakecluster1
    namespace: mig

  destMigClusterRef:
    name: doesnotexist
    namespace: mig
[...]
```

**Resulting SIGSEGV**
```
E0809 18:32:08.272555       1 runtime.go:69] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
/go/src/github.com/fusor/mig-controller/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:76
/go/src/github.com/fusor/mig-controller/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
/go/src/github.com/fusor/mig-controller/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
/usr/local/go/src/runtime/asm_amd64.s:573
/usr/local/go/src/runtime/panic.go:502
/usr/local/go/src/runtime/panic.go:63
/usr/local/go/src/runtime/signal_unix.go:388
/go/src/github.com/fusor/mig-controller/pkg/controller/migplan/validation.go:472
/go/src/github.com/fusor/mig-controller/pkg/controller/migplan/migplan_controller.go:201
/go/src/github.com/fusor/mig-controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:215
/go/src/github.com/fusor/mig-controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:158
/go/src/github.com/fusor/mig-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
/go/src/github.com/fusor/mig-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
/go/src/github.com/fusor/mig-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
/usr/local/go/src/runtime/asm_amd64.s:2361
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x128 pc=0x1482ce2]

goroutine 288 [running]:
github.com/fusor/mig-controller/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
        /go/src/github.com/fusor/mig-controller/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x107
panic(0x16ccbe0, 0x270cab0)
        /usr/local/go/src/runtime/panic.go:502 +0x229
github.com/fusor/mig-controller/pkg/controller/migplan.ReconcileMigPlan.validatePvSelections(0x1ae43c0, 0xc420684060, 0xc4202d1180, 0xc420411860, 0x0, 0x0)
        /go/src/github.com/fusor/mig-controller/pkg/controller/migplan/validation.go:472 +0x282
github.com/fusor/mig-controller/pkg/controller/migplan.(*ReconcileMigPlan).Reconcile(0xc4206c18c0, 0xc420076fb8, 0x3, 0xc420076fc0, 0xe, 0x2721c40, 0x2, 0x3, 0xc420180000)
        /go/src/github.com/fusor/mig-controller/pkg/controller/migplan/migplan_controller.go:201 +0x2f4
github.com/fusor/mig-controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc420181540, 0x0)
        /go/src/github.com/fusor/mig-controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:215 +0x188
github.com/fusor/mig-controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1()
        /go/src/github.com/fusor/mig-controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:158 +0x36
github.com/fusor/mig-controller/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc4364ff200)
        /go/src/github.com/fusor/mig-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x54
github.com/fusor/mig-controller/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc4364ff200, 0x3b9aca00, 0x0, 0x271f701, 0xc4205ab4a0)
        /go/src/github.com/fusor/mig-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134 +0xbd
github.com/fusor/mig-controller/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc4364ff200, 0x3b9aca00, 0xc4205ab4a0)
        /go/src/github.com/fusor/mig-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by github.com/fusor/mig-controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start
        /go/src/github.com/fusor/mig-controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:157 +0x35b
```

--- a/pkg/controller/migplan/validation.go
+++ b/pkg/controller/migplan/validation.go
@@ -481,6 +481,9 @@ func (r ReconcileMigPlan) validatePvSelections(plan *migapi.MigPlan) error {
 		log.Trace(err)
 		return err
 	}
+	if destMigCluster == nil {
+		return nil
+	}
 	for _, storageClass := range destMigCluster.Spec.StorageClasses {
 		storageClasses[storageClass.Name] = storageClass.AccessModes
 	}

